### ISSUE TITLE
pd.merge() doesn't support (named) Series

### ISSUE THREAD

**Issue by ecoughlan** [modin-project/modin#877]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04.2
- **Modin installed from (source or binary)**: pip
- **Modin version**: 0.6.3
- **Python version**: 3.7.3
- **Exact command to reproduce**: See below

### Describe the problem
This is special-cased in pandas, and doesn't work with modin:
```python
pd.merge(pd.DataFrame(range(10), columns=["foo"]), pd.Series(range(10), name="foo"))
```

### Source code / logs
Possible fix: https://gist.github.com/ecoughlan/30197e6de8664c0ca670ab1aeb4b8d64

**Comment by devin-petersohn** [comment]
Thanks for posting @ecoughlan, the fix you posted in the gist would likely work. If you're willing to contribute it with some test cases to verify correctness, let me know!

**Comment by devin-petersohn** [comment]
Resolved via #878
